File: Skills.Shared/V1/Request/CharacterRequest.cs
using System.ComponentModel.DataAnnotations;

namespace Skills.Shared.V1.Request;

public class CharacterRequest
{
    public int Priority { get; init; }
    [MaxLength(100)]
    public string BuildName { get; init; }

    public string StartingDate { get; set; }
    [MaxLength(1000)]
    public string? Story { get; set; }


    public Guid? PhotoId { get; set; }
    public Guid SkillSetId { get; set; }

    [MaxLength(8)]
    public IEnumerable<CharacterSkillRequest> Skills { get; init; }

}

File: Skills/ClientApp/src/models/RequestModels/CharacterRequest.ts
import FileModel from "../FileModel"
import SkillRequest from "./SkillRequest"

export default interface CharacterRequest {
    id?: string
    priority: number
    buildName: string
    startingDate: Date
    story?: string
    photoId?: string
    photo?: FileModel
    skillSetId:string
    skills: SkillRequest[]
}